opencog/attention/experimental/HebbianUpdatingAgent.cc

```diff
@@ -52,7 +57,12 @@ void HebbianUpdatingAgent::run()
     std::back_insert_iterator< std::vector<Handle> > out_hi(atoms);
 
     _as->get_handle_set_in_attentional_focus(out_hi);
-
+     /*Without adding this sleep code right below the above method call,
+     * nlp-parse evaluation thread waits for minutes before it gets a chance to
+     * run.
+     */
+    std::this_thread::sleep_for(std::chrono::nanoseconds(1));
```

Comment:
`std::this_thread::yield()` should accomplish the same thing.

This is an interesting issue. It implies that the HebbbianUpdatingAgent is pounding the crap out of the atomspace.  Probably here: https://github.com/opencog/atomspace/blob/master/opencog/atoms/base/Atom.cc#L196  

The importance index shares a common lock in the atomspace with atom insertion/deletion, so updating the importance, via the attention agents, will make everyone who creates atoms go slower. This suggests that a re-design is in order.